### DISCUSSION TITLE
Allow backoffice users to toggle between the default and custom view of a block list property.

### DISCUSSION THREAD

**Discussion by markadrake** [umbraco/Umbraco-CMS#11790] [Features and ideas]
As it stands now, you can apply a custom view to a block type. When editing content - if a block type has a custom view, that custom view is rendered.

I often find myself wishing for a toggle (while editing content) that would allow me to turn off custom views temporarily so I can more easily find the content I'm looking for.

What I'm proposing here is a set of icons [default|custom] that would allow any content editor to toggle the view presented.

The default can remain [custom] so if a view has a custom html template assigned – that will be used. OOTB so far.

But if you were to select [default] it would turn these custom html templates off (temporarily). This is the new feature I'm asking for.

Any opinions / recommendations / support from the community?

**Tracking**
- [ ] Is a property action the best UI/UX?
- [ ] Is (x) the best icon?
- [ ] Is "Toggle Custom Views On/Off" the best EN / EN-US text?
- [ ] CS translation
- [ ] CY translation
- [ ] DA translation
- [ ] EN translation
- [ ] EN-US translation
- [ ] FR translation
- [ ] IT translation
- [ ] NL translation
- [ ] TR translation

**Comment by nathanwoulfe** [comment]
Sounds like a useful addition - from memory the logic for determining custom or default view is pretty straightfoward so wouldnt be a super complex change. 

Can't immediately think of any reason not to do this...

**Comment by markadrake** [comment]
Here is a quick demo of this in action.

I'm wondering if I can get feedback from anyone at Umbraco HQ & community before I jump into creating an issue and a pull request?

https://user-images.githubusercontent.com/713355/149468936-001edf0f-bab9-4277-b858-c08354e8f942.mp4

**Comment by skttl** [comment]
I don't think editors should know about custom views and non custom views. My kneejerk reaction would be that your custom views are simply not fit for what you want to do.

I guess you want to get some kind of WYSIWYG experience, but becuase it is too overwhelming, you need to turn it off from time to time. In stead I think you should use the preview function - or if core could have a preview function similar to the one in Perplex Content Blocks.

**Reply by markadrake** [reply]
Hi there! Unfortunately your feedback leaves me with more questions than I originally had. Please bare with me - but what is the intention of the **custom view** functionality? The only purpose I can imagine is to provide a more accurate representation of what blocks look like to the content author. The fact it's built to show an external stylesheet as part of the shadow DOM seems like it's specifically geared to show a highly stylized component without affecting anything in the backoffice. I thought the sole purpose of the custom view was for WYSIWYG. :( 

**I am familiar with Preview Mode.** IMO it's a poor experience for content authors. It requires them to save the page, open up preview, and do this odd handshake to keep viewing the page. Preview mode breaks contextual editing and puts an overbearing and unnecessary process in place.

I'm not familiar with Perplex Content Blocks. I'll have to look that up! 

Here's a quick short page I put together. It only has a slider and a set of breadcrumbs at the moment. Bare in mind this is already starting to scroll.

![image](https://user-images.githubusercontent.com/713355/149491199-6acc947f-5562-42e9-894b-36376fd7e419.png)

![image](https://user-images.githubusercontent.com/713355/149491495-c8c57f8f-16e6-4116-aae7-23cce3afc03d.png)

I generally see 5-8 content blocks (some repeatable) on most pages my clients create. It's not unwieldly, nor unmanageable. Their are some trade-offs to using custom views IMO. When your blocks are shown as custom views it's (1) hard to identify when one block begins and ends, (2) because they run in together it gives no hint or clue that they can still be clicked, dragged, and sorted. My clients never think this is possible until I ask them to take a metaphorical leap of faith (just click and drag - trust me!)

It seems more obvious and more natural to switch to the other view if you are (1) scanning for a specific content block on the page, and (2) if you desire to drag and reposition content blocks.

![image](https://user-images.githubusercontent.com/713355/149491939-fe463703-02d2-40c6-8a65-433775944a34.png)

I'd be interested in more feedback like this to understand what the intention of custom views are (for the developer) –vs– how they've actually been implemented (for editors).
